reflex/app.py

```diff
@@ -249,9 +249,9 @@ async def preprocess(self, state: State, event: Event) -> StateUpdate | None:
         """
         for middleware in self.middleware:
             if asyncio.iscoroutinefunction(middleware.preprocess):
-                out = await middleware.preprocess(app=self, state=state, event=event)
+                out = await middleware.preprocess(app=self, state=state, event=event)  # type: ignore
```

Comment:
Why are the type ignore's needed now? Can we updated the .pyi so we don't need them?

Reply:
I didn't find a way, because of the self reference, pyright distinguish between the App in app.py and the one in App.pyi (they are considered different object, so self referenced doesn't pass pyright)